jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpDestination.java

```diff
@@ -489,15 +484,22 @@ public void release(Connection connection)
         else
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("{} is stopped", client);
+                LOG.debug("Closing {}, {} is stopped", connection, client);
             connection.close();
         }
     }
 
+    private boolean releaseConnection(Connection connection)
+    {
+        boolean released = connectionPool.release(connection);
+        if (!released)
+            connection.close();
+        return released;
+    }
```

Comment:
This would be clearer:
```suggestion
    {
        boolean close = connectionPool.release(connection);
        if (close)
            connection.close();
        return close;
    }
```

Reply:
We cannot change the meaning of the return value of `ConnectionPool.release()`.